src/testing/deploy/upgrade_test.go

```diff
@@ -136,32 +143,46 @@ func TestUpgradeTrigger(t *testing.T) {
 					return errors.New("not ready")
 				}
 				return nil
-			})
+			}
+			verifyCommitPost28x := func () error {
+				fmt.Printf("verifyCommitPost28x running with version %s", from)
+				commits, err := c.ListCommit(client.NewRepo(pfs.DefaultProjectName, "TestTrigger2"), nil, nil, 0)
+				require.NoError(t, err)
+				require.Equal(t, 1, len(commits))
```

Comment:
I think there should be more than one commit here, the test just sees 1 commit because it's not done yet. I think rather than having different verification functions on each version we can change `aliasCI.Commit.Id != latestDataCI.Commit.Id` in the retry to something like `listCommitCount < expectedCommitCount` for all versions. Based on testing it seemed like the commit count was the same across versions. I think it's nice to have the same logic verifying all versions.

Reply:
okay let me see what i can pull off 👍